src/cdk/collections/selection-model.ts

```diff
@@ -90,6 +102,9 @@ export class SelectionModel<T> {
    * Determines whether a value is selected.
    */
   isSelected(value: T): boolean {
+    if (this._compareWith) {
+      return [...this._selection].some(v => this._compareWith!(v, value));
```

Comment:
Since this is in a getter, it might be better to use a regular loop instead of converting it into an array just so it can use `some`.